### DISCUSSION TITLE
Database error after update

### DISCUSSION THREAD

**Discussion by lifeoflejf** [amidaware/tacticalrmm#1209] [General]
Hi!

I've updated from 0.13.4 to 0.14.2, and in the update script I'm getting a bunch of database permission errors:
psycopg2.errors.InsufficientPrivilege: permission denied for table django_migrations

I rebooted & tried again with -force, and i get something similar:

![image](https://user-images.githubusercontent.com/73968762/178841943-93d47565-2421-448e-a1c6-196664bce12f.png)

if i run psql i get an error that the role tatical does not exist:
![image](https://user-images.githubusercontent.com/73968762/178842027-aa4174ec-a979-4898-9aa2-6789c0f8463d.png)

meshcentral still works 100%
any ideas?

**Comment by silversword411** [comment]
issues are for bugs, this doesn't sound like a bug. For troubleshooting, you should join Discord.

If you're getting permission errors, sounds like there's a problem with your server setup, user you're running from `tactical` should have sudo privileges.

**Comment by bbrendon** [comment]
Do you have grafana installed?

**Comment by wh1te909** [comment]
this would only happen if you were messing around with your database in an unsupported way

**Comment by lifeoflejf** [comment]
I appreciate all your help. 

I didn't install grafana, I've only used the update script over the past year or so. I'm also not smart enough to touch the database :(

Afaik the tactical account does have sudo access, since it asks for a login at the beginning of the script and then goes though with apt update.

Best to backup/restore at this point?

**Reply by wh1te909** [reply]
run the following script to try and fix db permissions then try update with force again
```bash
#!/usr/bin/env bash

pgusername=$(grep -w USER /rmm/api/tacticalrmm/tacticalrmm/local_settings.py | sed 's/^.*: //' | sed 's/.//' | sed -r 's/.{2}$//')

sudo -u postgres psql -c "ALTER ROLE ${pgusername} SET client_encoding TO 'utf8'"
sudo -u postgres psql -c "ALTER ROLE ${pgusername} SET default_transaction_isolation TO 'read committed'"
sudo -u postgres psql -c "ALTER ROLE ${pgusername} SET timezone TO 'UTC'"
sudo -u postgres psql -c "GRANT ALL PRIVILEGES ON DATABASE tacticalrmm TO ${pgusername}"
```

**Comment by silversword411** [comment]
A log of the full update that led to this in a DM (Direct Message - might have private info) thru Discord to one of us might help determine why it happened in the first place.
